src/Module/Directory.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\Content\Nav;
+use Friendica\Content\Pager;
+use Friendica\Content\Widget;
+use Friendica\Core\Hook;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Model\Contact;
+use Friendica\Model\Profile;
+use Friendica\Util\Proxy as ProxyUtils;
+use Friendica\Util\Strings;
+
+/**
+ * Shows the local directory of this node
+ */
+class Directory extends BaseModule
+{
+	public static function init()
+	{
+		$app = self::getApp();
+
+		if (local_user()) {
+			$app->page['aside'] .= Widget::findPeople();
+			$app->page['aside'] .= Widget::follow();
+		} else {
+			unset($_SESSION['theme']);
+			unset($_SESSION['mobile-theme']);
+		}
+	}
+
+	public static function post()
+	{
+		if (!empty($_POST['search'])) {
+			self::getApp()->data['search'] = $_POST['search'];
```

Comment:
Please remove this block and use `$_REQUEST['search']` later instead.

This means you can merge `init()` with `content()` as well and remove this nasty dependency on `App->data` that you could have converted to a static class variable if it was really needed.